enclave.go

```diff
@@ -529,27 +602,25 @@ func (e *Enclave) setCertFingerprint(rawData []byte) error {
 	return nil
 }
 
-// SetKeyMaterial registers the enclave's key material (e.g., secret encryption
-// keys) as being ready to be synchronized to other, identical enclaves.  Note
-// that the key material's underlying data structure must be marshallable to
-// JSON.
-//
-// This is only necessary if you intend to scale enclaves horizontally.  If you
-// will only ever run a single enclave, ignore this function.
-func (e *Enclave) SetKeyMaterial(keyMaterial any) {
-	e.Lock()
-	defer e.Unlock()
-
-	e.keyMaterial = keyMaterial
+// getLeader returns the leader enclave's URL.
+func (e *Enclave) getLeader(path string) *url.URL {
```

Comment:
What's the expected behavior if the leader shutsdown? Will it restart itself on the same place such that shouldn't cause a failure here and would we be alerted if this occurs?

Reply:
Right, if the leader terminates for whatever reason, Kubernetes it meant to restart it.

Reply:
Cool, I like this approach to keep it simple.